packages/s3-credentials-endpoint/main.tf

```diff
@@ -0,0 +1,161 @@
+data "aws_caller_identity" "current" {}
+
+resource "aws_dynamodb_table" "access_tokens" {
+  name         = "${var.prefix}-s3-credentials-access-tokens"
+  billing_mode = "PAY_PER_REQUEST"
+  hash_key     = "accessToken"
+
+  attribute {
+    name = "accessToken"
+    type = "S"
+  }
+}
+
+data "aws_iam_policy_document" "assume_lambda_role" {
+  statement {
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role" "s3_credentials_lambda" {
+  name                 = "${var.prefix}-S3CredentialsLambda"
+  assume_role_policy   = data.aws_iam_policy_document.assume_lambda_role.json
+  permissions_boundary = var.permissions_boundary
+}
+
+data "aws_iam_policy_document" "s3_credentials_lambda" {
+  statement {
+    actions   = ["lambda:InvokeFunction"]
+    resources = [var.sts_credentials_lambda_arn]
+  }
+
+  statement {
+    actions   = ["dynamodb:GetItem"]
+    resources = [aws_dynamodb_table.access_tokens.arn]
+  }
+
+  statement {
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents"
+    ]
+    resources = ["arn:aws:logs:*:*:*"]
+  }
+  statement {
+    actions = [
+      "ec2:CreateNetworkInterface",
+      "ec2:DescribeNetworkInterfaces",
+      "ec2:DeleteNetworkInterface"
+    ]
+    resources = ["*"]
+  }
+}
+
+resource "aws_iam_role_policy" "s3_credentials_lambda" {
+  policy = data.aws_iam_policy_document.s3_credentials_lambda.json
+  role   = aws_iam_role.s3_credentials_lambda.id
+}
+
+resource "aws_lambda_function" "s3_credentials" {
+  function_name = "${var.prefix}-s3-credentials-endpoint"
+  # TODO Fetch this from ... somewhere. Or package it with the module zip file? Probably the better option
```

Comment:
This comment can probably be removed.

Reply:
https://github.com/nasa/cumulus/pull/1086/commits/1af739dea0a39b44c85d33dd9768f3e93f698d15